external-plugins/rehearse/plugin/server/eventsserver.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"net/http"
+
+	v1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"k8s.io/test-infra/prow/github"
+	"kubevirt.io/project-infra/external-plugins/rehearse/handler"
+)
+
+type GitHubEventsServer struct {
+	prowClient     v1.ProwJobInterface
+	tokenGenerator func() []byte
+	eventsChan     chan<- *handler.GitHubEvent
+}
+
+func NewGitHubEventsServer(
+	tokenGenerator func() []byte,
+	eventsChan chan<- *handler.GitHubEvent) *GitHubEventsServer {
+
+	return &GitHubEventsServer{
+		tokenGenerator: tokenGenerator,
+		eventsChan:     eventsChan,
+	}
+}
+
+func (s *GitHubEventsServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+
+	eventType, eventGUID, eventPayload, eventOk, _ := github.ValidateWebhook(writer, request, s.tokenGenerator)
+
+	if !eventOk {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	s.eventsChan <- &handler.GitHubEvent{
```

Comment:
Your comment caused me to re-think that I did and actually question why I even used a channel here and I don't remember that I had a good reason for it except for separation of concerns between the events handler and the events server itself, but I assume that by making the handler a member of the events server instead of the channel still solves this problem and even simplifies the testing.